app/utils/clone-learner-group.js

```diff
@@ -0,0 +1,32 @@
+import RSVP from 'rsvp';
+
+const { map } = RSVP;
+
+export default async function cloneLearnerGroup(store, group, cohort, withLearners, parent = null) {
+  let newGroup = store.createRecord('learner-group', group.getProperties(
+    'title',
+    'location'
+  ));
+  newGroup.set('cohort', cohort);
+  if (parent) {
+    newGroup.set('parent', parent);
+  }
+  if (withLearners) {
+    const users = await group.get('users');
+    await map(users.toArray(), async user => {
+      await newGroup.addUserToGroupAndAllParents(user);
+    });
+  }
+  const instructors = await group.get('instructors');
+  newGroup.set('instructors', instructors);
+  const children = await group.get('children');
+  let newChildren = await map(children.toArray(), async child => {
+    return await cloneLearnerGroup(store, child, cohort, withLearners, newGroup);
+  });
+  let flat = newChildren.reduce((flattened, obj) => {
+    return flattened.pushObjects(obj.toArray());
+  }, []);
+  const newGroups = [].concat([newGroup], flat);
```

Comment:
just return here, no need for variable `newGroups`.
change to: `return [].concat([newGroup], flat);`